src/System.IO.UnmanagedMemoryStream/tests/UmaTests.cs

```diff
@@ -23,6 +23,13 @@ public static void UmaInvalidReadWrite()
                 Assert.Throws<ArgumentOutOfRangeException>(() => uma.Write(capacity, 12345));
                 Assert.Throws<ArgumentException>(() => uma.Write(capacity - 1, 0.123));
 
+                // UMA should throw when reading bad decimal values (some bits of flags are reserved and must be 0)
+                uma.Write(0, 0); // lo
```

Comment:
If pointer has a small value (including null of course!), that certainly could show up as NullReferenceException.